### ISSUE TITLE
jinzhu/gorm: Ignore some errors like record not found

### ISSUE THREAD

**Issue by moonsub-kim** [DataDog/dd-trace-go#805]
I wish ignore errors especially record not found that is unnecessary to classify as errors in my server.
![image](https://user-images.githubusercontent.com/13393411/103546755-d8b33580-4ee6-11eb-8285-3bfb22824626.png)
The server queries by request from the client that can manipulate req body, then we always allow record_not_found error.
Most of record_not_found error makes me indifferent to see error rate dashboard.

To Extendability,I suggest adding an option that can ignore some type of errors like below
```go
WithErrorFilter(func(err error) bool{
    if errors.Is(err, gorm.ErrRecordNotFound) {
        return false
    }
    return true
})
```

If you agree about it, I will make a PR

**Comment by knusbaum** [comment]
@moonsub-kim

This seems like an acceptable option to add. We added something similar to `contrib/go-chi/chi` recently (#773). 
In that one, we ended up calling the option `WithStatusCheck`, so maybe for consistency we can call this `WithErrorCheck`

We also [just merged `contrib/gorm.io/gorm.v1`](#759) which is where the next version of gorm [moved](https://github.com/jinzhu/gorm/blob/master/README.md). We should add the option there too to keep parity between gorm integration versions.
